backend/apps/readings/analysis.py

```diff
@@ -369,7 +370,68 @@ def compute_median_view_time(self):
         else:
             list_of_times.sort()
             median_view_time = statistics.median(list_of_times)
-        return round(median_view_time)
+        return median_view_time
+
+    def compute_reread_counts(self, question, context):
+        """"
+        Given a list of student response dicts,
+        return a dictionary containing the number of times students had to reread the text
+        :param question: string, question for which reread counts is collected
+        :param context: string, context for which reread counts is collected
+        :return: tuple, which contains the question, context, and number of students that reread
+         the text 0 times, 1 time, etc.
+        """
+
+        # Checks that the question and context are not blank
+        if question == '' or context == '':
+            return {}
+
+        # Collects the reread count for every student id of the provided context and question
+        raw_reread_counts = []
+        for response in self.responses:
+            table_context = response.context.text
+            table_question = response.question.text
+            view_count = len(response.get_parsed_views())
+            if context in table_context:
+                if question in table_question:
+                    raw_reread_counts.append(view_count)
+
+        # Tallies the raw reread counts into the dictionary to be returned
+        organized_data = {
```

Comment:
Why not construct these keys dynamically, so you can count every single possible number of reread counts?